### PR TITLE
Adding support for Smartwatch Amazfit Pace/Stratos

### PR DESCRIPTION
**Pull request by Klaus3d3**
for the watch app see https://github.com/Klaus3d3/XdripWidgetForAmazfit-Klaus

### CODE REVIEW

**Comment by Klaus3d3** [comment]
This changes add an additional service to communicate with the Xiaomi Huami Amazfit Pace & Stratos Smartwatches.
For now it offers native push of glucose values to the widget installed on watch.
Future features i'm working on: pushing snoozeable alarms, HRM and stepp transmission to phone, also Treatment entering should be possible, if i find the Interface within xDrip.


These smartwatches offer a custom android based OS, a semitransflective Display, and therefore a battery life of 4 to 6 days. IP67 for Pace and IP68 for Stratos, HRM, GPS, internal Musciplayer, and many sport modes. Since the library for communication is now open for developers, many things will come.

File: local.properties [hunk 1]
```diff
@@ -0,0 +1,8 @@
+## This file must *NOT* be checked into Version Control Systems,
+# as it contains information specific to your local configuration.
+#
+# Location of the SDK. This is only used by Gradle.
+# For customization when using a Version Control System, please read the
+# header note.
+#Tue May 01 20:07:02 CEST 2018
+sdk.dir=C\:\\Android_Settings
```
Comment (dabear) [inline local.properties]: This file should be removed

**Comment by jamorham** [comment]
Hi, thanks for this, can you provide a photo of how it looks on the watch.

File: build.gradle [hunk 2]
```diff
@@ -9,11 +9,13 @@ buildscript {
     dependencies {
```
Comment (jamorham) [inline build.gradle]: Please remove this file from the commit as there are no changes to it related to the PR

File: xDrip-master.iml [hunk 3]
```diff
@@ -0,0 +1,31 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (jamorham) [inline xDrip-master.iml]: Please remove this file from the PR, it is local only. I thought there was .gitignore for this already. Thanks

File: local.properties (hunk 1 above)
Comment (Klaus3d3) [reply local.properties]: https://ibb.co/jFjrkd

Sure

**Comment by Klaus3d3** [comment]
https://ibb.co/jFjrkd

Photo of the watch widget.

For the next weeks i'm on hollydays. Don't know if i find time and computer access to do the changes.

File: xDrip-master.iml (hunk 3 above)
Comment (Klaus3d3) [reply xDrip-master.iml]: Sorry, being a noop in this. Shall i delete both files completely?

**Comment by Klaus3d3** [comment]
hopefully i've done all the needed changes to the PR now.

**Comment by Klaus3d3** [comment]
Next week i will have time for this. But then i definately need help to get this PR straight. I don't get what to do.

**Comment by Klaus3d3** [comment]
The library comes from here:
https://github.com/KieronQuinn/AmazfitCommunication

Kierons Modification are open source if i read it right.

And i got the approval of HUAMIs Public Relation manager Pavel Zeldin for the reverse engineered part. 
HUAMI seems to be very dev friendly.

**Comment by Klaus3d3** [comment]
Sorry, i'm not sure about the approval from Pavel. I was to quick. I will get an answer about this tomorrow.

**Comment by jamorham** [comment]
Ok please check approvals and if we can get the jar file from the official source via a gradle include that would be ideal. 

For the build.gradle issue, take the current master version and paste it in to yours where it appears to be deleted according to this pr: https://github.com/NightscoutFoundation/xDrip/blob/master/build.gradle then commit that change and push to this branch. It should result in the changes to build.gradle vanishing which is what we want.

Other conflicts may be more fiddly to resolve. I can take a look at those perhaps.

**Comment by jamorham** [comment]
Slightly unrelated question, would this work with AMAZFIT Bip? I don't know anything about this range really but noticed that one had a lower price.

**Comment by Klaus3d3** [comment]
I know the bip seems very promissing due to its 30day battery life plus sam display tech like pace & stratos. But sadly it has a different OS and has less powerful hardware. So NO, it does not work.
I don't have a bip, but i know there are very powerful 3rd party companion apps and there are also different roms for the bip. 
Maybe something is possible, but not with this library.

**Comment by Klaus3d3** [comment]
Maybe if fixed the build gradle but don't know if i have done it right.

**Comment by jamorham** [comment]
Ok, I think this is getting in to more of a mess now. Is it okay if I attempt to fix the problems and push to your repository?

**Comment by Klaus3d3** [comment]
for god's sake, Yes please, fix it. Obviously i don't know what i'm doing. :-(

**Comment by jamorham** [comment]
Please review the new version here: https://github.com/NightscoutFoundation/xDrip/pull/477

**Pull request merged**